src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_SIZE_DATA = 10;
+    public static final double INCREASE_BY = 1.5;
+    private int size;
+    private T[] dataType;
+
+    public ArrayList() {
+        dataType = (T[]) new Object[DEFAULT_SIZE_DATA];
+    }
+
     @Override
     public void add(T value) {
-
+        getArrayList();
+        dataType[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
 
+        getIndexFrom(index);
+        getArrayList();
+        System.arraycopy(dataType,index,dataType,index + 1, size - index);
+        dataType[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        getIndex(index);
+        return dataType[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        getIndex(index);
+        dataType[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        getIndex(index);
+        T deleteArray = dataType[index];
+        size--;
+        System.arraycopy(dataType,index + 1, dataType,index, size - index);
+        return deleteArray;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int j = 0; j < size; j++) {
+            if (element != null
+                    && element.equals(dataType[j])
+                    || element == dataType[j]) {
+                return remove(j);
+            }
+        }
+        throw new NoSuchElementException("Element not found" + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void getIndex(int index) {
```

Comment:
method name is a bit misleading - better to change it for example to indexValidation()